### PR TITLE
all-your-base: add to track

### PR DESCRIPTION
**Pull request by stkent**
A port of the [xjava implementation](https://github.com/exercism/xjava/tree/master/exercises/all-your-base/src) of this exercise.

[Canonical data](https://github.com/exercism/x-common/blob/39de23b0de86f1be018c5b62d1dd5b952349f286/exercises/all-your-base/canonical-data.json).

### CODE REVIEW

**Comment by stkent** [comment]
I couldn't figure out why including a message in the `@Ignore` annotation caused test compilation to fail (it works fine in the Java track). If anyone in @exercism/kotlin knows why, I'd love to know!

File: exercises/all-your-base/src/example/kotlin/BaseConverter.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            result[largestExponent - currentExponent] = coefficient
+
+            remainder -= (coefficient * Math.pow(newBase.toDouble(), currentExponent.toDouble())).toInt()
+        }
+
+        return result
+    }
+
+    private fun validateInputs(originalBase: Int, originalDigits: IntArray) {
+        if (originalBase < MINIMUM_VALID_BASE) {
```
Comment (sdavids13) [inline exercises/all-your-base/src/example/kotlin/BaseConverter.kt]: All of these if/throw statements can be simplified by using Kotlin's `require` method, see an example here: https://github.com/exercism/xkotlin/blob/02ecc5a70719f0b0e741992f33879f880f7c407f/exercises/pascals-triangle/src/example/kotlin/PascalsTriangle.kt#L4

Comment (stkent) [inline exercises/all-your-base/src/example/kotlin/BaseConverter.kt]: ❤️ it. Updated!

**Comment by sdavids13** [comment]
Looks good, thanks for the contributions. As for the comment within the `@Ignore` I'm not sure what the problem may be, it seems to work on my local machine for some reason.

**Pull request merged**